sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc

```diff
@@ -0,0 +1,508 @@
+
+a|
+[source,c++]
+----
+awidth<width>
+----
+a|
+Specifies the width of the memory-mapped address bus in bits. The default is
+set to 64.
+
+This property is only meaningful for pointer kernel arguments and only
+when the `buffer_location` property is specified.
+
+a|
+[source,c++]
+----
+dwidth<width>
+----
+a|
+Specifies the width of the memory-mapped data bus in bits. The default is set
+to 64.
+
+This property is only meaningful for pointer kernel arguments and only
+when the `buffer_location` property is specified.
+
+a|
+[source,c++]
+----
+read_write_mode<mode>
+----
+a|
+Specifies the port direction of the memory interface associated with the input
+pointer. `mode` can be one of:
+
+`read_write` - Interface can be used for read and write operations.
+
+`read` - Interface can only be used for read operations.
+
+`write` - Interface can only be used for write operations.
+
+The default is set to `read_write`.
+
+For convenience, the following are provided:
+
+ - read_write_mode_read
+ - read_write_mode_write
+ - read_write_mode_readwrite
+
+This property is only meaningful for pointer kernel arguments and only
+when the `buffer_location` property is specified.
+
+a|
+[source,c++]
+----
+latency<value>
+----
+a|
+Specifies the guaranteed latency in cycles, from when a read command exits
+the kernel to when the external memory returns valid read data. The default
+is set to 1.
+
+A value of 0 specifies a variable latency and a positive value specifies a
+fixed latency.
+
+This property is only meaningful for pointer kernel arguments and only
+when the `buffer_location` property is specified.
+
+a|
+[source,c++]
+----
+maxburst<value>
+----
+a|
+Specifies the maximum number of data transfers that can be associated with a
+read or write transaction. The default is set to 1.
+
+This property is only meaningful for pointer kernel arguments and only
+when the `buffer_location` property is specified.
+
+a|
+[source,c++]
+----
+wait_request<flag>
+----
+a|
+Specifies whether the 'wait request' signal is generated or not. This signal is
+asserted by the memory system when it is unable to respond to a read or write
+request. The default is set to `false`.
+
+For convenience, the following are provided:
+
+ - wait_request_requested
+ - wait_request_not_requested
+
+This property is only meaningful for pointer kernel arguments and only
+when the `buffer_location` property is specified.
+|===
+--
+
+=== Usage Examples
+
+The example below shows a simple kernel with one `annotated_ptr` kernel
+argument and one `annotated_arg` kernel argument.
+
+.Usage Example
+```c++
+using sycl::ext::oneapi::experimental;
+{
+  sycl::queue q{...};
+
+  // Allocate memory
+  auto ptr_a = ...
+  constexpr int kN = 10;
+
+  // Add properties
+  auto arg_a = annotated_ptr(ptr_a, properties{
+    register_map, buffer_location<1>, awidth<18>, dwidth<64>});
+  auto arg_n = annotated_ptr(kN, properties{register_map, stable});
```

Comment:
The text above states that one of these is an annotated_arg but both are annotated_ptrs.